test/utils.py

```diff
@@ -137,6 +137,9 @@ def expect_string(pattern, logfile_path):
     with open(logfile_path) as file, mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as log:
         return re.search(pattern, log)
 
+def expect_string_esc(pattern, logfile_path):
```

Comment:
On the one had I like the idea, because I was also thinking that the regexps are a bit too much. Before the `pytest` migration, everything was done using grep, which uses BREs. This is the reason the `[` and `]` were escaped, but the `(` and `)` weren't.

On the other hand, we're kind of making up our own syntax.

Reply:
I add this function just for simplify tests syntax (reading and writing - you can just pass strong from log). It is not mandatory but can be useful. 
If this some new syntax... maybe but to be honest I don't see many profits of using full regex in log pattern check. It is almost always `some_fixed_text` or `some_fixed_text `**`file_path`**` fixed_line_number` and regex or wildcard is used only for file_path